zkir/Dialect/Field/Conversions/ExtFieldToLLVM/ExtFieldToLLVM.cpp

```diff
@@ -55,18 +65,18 @@ struct ConvertExtFromCoeffs : public ConvertOpToLLVMPattern<ExtFromCoeffsOp> {
   LogicalResult
   matchAndRewrite(ExtFromCoeffsOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto loc = op.getLoc();
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
     auto structType = typeConverter->convertType(op.getType());
-    if (isa<QuadraticExtFieldType>(op.getType())) {
-      if (adaptor.getInput().size() != 2)
-        return op.emitOpError(
-            "expected two input types for quadratic extension field");
-      auto extFieldStruct = SimpleStructBuilder<2>::initialized(
-          rewriter, loc, structType, adaptor.getInput());
-      rewriter.replaceOp(op, {extFieldStruct});
-      return success();
-    }
-    return op.emitOpError("unsupported output type");
+
+    auto extField = dyn_cast<ExtensionFieldTypeInterface>(op.getType());
+    if (!extField)
+      return op.emitOpError("unsupported output type");
```

Comment:
If the verification is needed, i think we should add this in `ExtFromCoeffsOp::verify()` later with `hasVerifier=1`!

```suggestion
    auto extField = cast<ExtensionFieldTypeInterface>(op.getType());
```